### PR TITLE
Bring master branch changes to to read_cache_release branch

### PR DESCRIPTION
**Pull request by gargnitingoogle**
### Description
This is an intermediate placeholder PR. It'll get more changes once https://github.com/GoogleCloudPlatform/gcsfuse/pull/1530 is merged.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - presubmit

### CODE REVIEW

File: tools/integration_tests/run_tests_mounted_directory.sh [hunk 1]
```diff
@@ -85,9 +85,11 @@ GODEBUG=asyncpreemptoff=1 go test ./tools/integration_tests/operations/...  -p 1
 sudo umount $MOUNT_DIR
 
 # Run tests with config "file-cache: max-size-in-mb" static mounting.
-echo "file-cache:
+echo "write:
```
Comment (gargnitingoogle) [inline tools/integration_tests/run_tests_mounted_directory.sh:88]: @ashmeenkaur @sethiay this echo command was the only source of conflict in this merge. Please confirm that this merge will work.

Comment (ashmeenkaur) [reply tools/integration_tests/run_tests_mounted_directory.sh:88]: @sethiay should we run these separately for different configs or is it fine to run them together?

**Comment by gargnitingoogle** [comment]
Integration and perf presubmits passed: [log](https://fusion2.corp.google.com/ci/kokoro/prod:gcsfuse%2Fgcp_ubuntu%2Fpresubmits%2Fperf_tests%2Fpresubmit/activity/3819babd-5c66-4572-ade3-92ebc5f74315/log)

File: tools/integration_tests/run_tests_mounted_directory.sh (hunk 1 above)
Comment (sethiay) [reply tools/integration_tests/run_tests_mounted_directory.sh:88]: Ideally, we should run them separately.

Comment (gargnitingoogle) [reply tools/integration_tests/run_tests_mounted_directory.sh:88]: Done.

**Comment by gargnitingoogle** [comment]
@ashmeenkaur @sethiay I resolved the only open issue. Please review and approve.

**Comment by sethiay** [comment]
> @ashmeenkaur @sethiay I resolved the only open issue. Please review and approve.

Thanks. LGTM.

Since most of the changes are in test scripts/CI setup. @Tulsishah can you please quickly sanity check this PR ?

**Comment by Tulsishah** [comment]
> > @ashmeenkaur @sethiay I resolved the only open issue. Please review and approve.
> 
> Thanks. LGTM.
> 
> Since most of the changes are in test scripts/CI setup. @Tulsishah can you please quickly sanity check this PR ?

Looks good to me

**Review by sethiay — APPROVED** [review]

**Pull request merged**